packages/flutter_tools/lib/src/custom_devices/custom_device_config.dart

```diff
@@ -94,117 +96,91 @@ class CustomDeviceConfig {
     required this.runDebugCommand,
     this.forwardPortCommand,
     this.forwardPortSuccessRegex,
-    this.screenshotCommand
-  }) : assert(forwardPortCommand == null || forwardPortSuccessRegex != null),
-       assert(
-         platform == null
-         || platform == TargetPlatform.linux_x64
-         || platform == TargetPlatform.linux_arm64
-       );
+    this.screenshotCommand,
+    this.embedderName,
+    this.configureNativeProject,
+    this.buildNativeProject,
+  })  : assert(forwardPortCommand == null || forwardPortSuccessRegex != null),
+        assert(platform == null ||
+            platform == TargetPlatform.linux_x64 ||
+            platform == TargetPlatform.linux_arm64);
 
   /// Create a CustomDeviceConfig from some JSON value.
   /// If anything fails internally (some value doesn't have the right type,
   /// some value is missing, etc) a [CustomDeviceRevivalException] with the description
   /// of the error is thrown. (No exceptions/errors other than JsonRevivalException
   /// should ever be thrown by this factory.)
   factory CustomDeviceConfig.fromJson(dynamic json) {
-    final Map<String, dynamic> typedMap = _castJsonObject(
-      json,
-      'device configuration',
-      'a JSON object'
-    );
+    final Map<String, dynamic> typedMap =
+        _castJsonObject(json, 'device configuration', 'a JSON object');
 
     final List<String>? forwardPortCommand = _castStringListOrNull(
-      typedMap[_kForwardPortCommand],
-      _kForwardPortCommand,
-      'null or array of strings with at least one element',
-      minLength: 1
-    );
+        typedMap[_kForwardPortCommand],
+        _kForwardPortCommand,
+        'null or array of strings with at least one element',
+        minLength: 1);
```

Comment:
can you update this and the other argument lists below split onto multilines to have a trailing comma after the last argument and then a newline before the close paren, per our style guide: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#formatting

Reply:
ok, did that (and reverted the formatting changes)